include/boost/url/decoded_view.hpp

```diff
@@ -0,0 +1,1012 @@
+    using const_reference = char;
+
+    /** The unsigned integer type used to represent size.
+    */
+    using size_type = std::size_t;
+
+    /** The signed integer type used to represent differences.
+    */
+    using difference_type = std::ptrdiff_t;
+
+    /** A read-only bidirectional iterator to the decoded range.
+
+        This is a read-only bidirectional iterator to
+        the decoded characters.
+
+     */
+#ifdef BOOST_URL_DOCS
+    using const_iterator = __see_below__;
+#else
+    class const_iterator;
+#endif
+
+    /// @copydoc const_iterator
+    using iterator = const_iterator;
+
+    /** Constructor
+
+        Default-constructed objects represent
+        the empty string.
+
+     */
+    BOOST_CONSTEXPR
+    decoded_view(
+        bool plus_to_space = true) noexcept
+        : p_(nullptr)
+        , n_(0)
+        , dn_(0)
+        , plus_to_space_(plus_to_space)
+    {}
+
+    /** Constructor
+     */
+    decoded_view(
+        char const* str,
+        bool plus_to_space = true) noexcept
+        : p_(str)
+        , n_(traits_type::length(str))
+        , dn_(pct_decode_bytes_unchecked(string_view(str)))
+        , plus_to_space_(plus_to_space)
+    {}
+
+    /** Constructor
+     */
+    decoded_view(
+        char const* str,
+        size_type len,
+        bool plus_to_space = true) noexcept
+        : p_(str)
+        , n_(len)
+        , dn_(pct_decode_bytes_unchecked(string_view(str, len)))
+        , plus_to_space_(plus_to_space)
+    {}
+
+    /** Constructor
+     */
+#ifndef BOOST_URL_DOCS
+    template <
+        class End,
+        typename std::enable_if<std::is_same<End, char const*>::value, int>::type = 0>
+    BOOST_CXX14_CONSTEXPR
+    decoded_view(
+        char const* first,
+        End last,
+        bool plus_to_space = true) noexcept
+#else
+    BOOST_CXX14_CONSTEXPR
+    decoded_view(
+        char const* first,
+        char const* last,
+        bool plus_to_space = true) noexcept
+#endif
+        : p_(first)
+        , n_(last - first)
+        , dn_(pct_decode_bytes_unchecked(string_view(first, last)))
+        , plus_to_space_(plus_to_space)
+    {
+        BOOST_ASSERT(last - first >= 0);
+    }
+
+    /** Constructor
+     */
+    decoded_view(
+        urls::string_view str,
+        bool plus_to_space = true) noexcept
+        : p_(str.data())
+        , n_(str.size())
+        , dn_(pct_decode_bytes_unchecked(str))
+        , plus_to_space_(plus_to_space)
+    {}
+
+#if !defined(BOOST_NO_CXX17_HDR_STRING_VIEW)
+    /** Constructor
+     */
+    decoded_view(
+        std::string_view str,
+        bool plus_to_space = true) noexcept
+        : p_( str.data() )
+        , n_( str.size() )
+        , dn_(pct_decode_bytes_unchecked(str))
+        , plus_to_space_(plus_to_space)
+    {}
+#endif
+
+    /** Returns an iterator to the beginning
+
+        Returns an iterator to the first decoded character of the string
+
+        @return const_iterator to the first character
```

Comment:
The "@return" clause is redundant here. How many times do we need to tell the user that this function returns an iterator? "C'mon man!"